lib/harmony/harmonytask/harmonytask.go

```diff
@@ -0,0 +1,386 @@
+package harmonytask
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/filecoin-project/lotus/lib/harmony/resources"
+	"github.com/gin-gonic/gin"
```

Comment:
In most other places we tend to use github.com/gorilla/mux where the default Golang mux/router isn't enough. No strong opinions, but would be slightly more consistent

Reply:
For consistency. Later, a modern replacement might be nice:
Gin:     
   f.GET("/"+name+"/:tID", func(c *gin.Context) {
      tID := c.Params("tID")
Gorilla: 
   f.Path("/"+name+"/{tID}").Methods("GET").HandlerFunc(func(w http.ResponseWriter, r *http.Request){
      tID := mux.Vars(r)["tID"]